### PR TITLE
feature(): Support get/set Adaptive Dynamic Range Compression, Dialogue Level

### PR DESCRIPTION
**Pull request by isolationism**

### CODE REVIEW

**Comment by wuub** [comment]
LGTM.

I'll merge it later today. Please feel free to remind me to do so if I forget :)

**Comment by wuub** [comment]
@isolationism my receiver does not support "Dialogue_Lvl" command, do you  think it would be reasonable to use something like:
`self.supports_method(self.zone, "Sound_Video", "Dialogue_Adjust", "Dialogue_Lvl")` 
to check if it's supported before doing a request and rise something more descriptive instead of `ResponseException`

On one hand "it's better to ask for forgiveness etc. etc" on the other doing a request when we know it will fail is a bit excessive :)  What do you think?

**Comment by isolationism** [comment]
Hmm, right before making the request is probably not the best time to do it — I would suggest that perhaps the best time to query features is actually when making the initial connection to the receiver, and download the full `desc.xml` file and parse that to create a map of what features are supported.

Of course that's a pretty involved solution, but maybe it's the right direction to be heading anyway. Eventually all of those XML tag strings are going to get pretty cumbersome.

Not sure if you'd prefer to make a quick patch in this case or just let this PR die, but I suspect there are probably a lot of features that are going to be specific to a particular device (beyond the basics like power and volume, anyway) so a long-term strategy is needed.

**Comment by wuub** [comment]
>  I would suggest that perhaps the best time to query features is actually when making the initial connection to the receiver, and download the full desc.xml file and parse that to create a map of what features are supported.

This is how `self.supports_method(...)` works underneath :) `desc_xml` is fetched  during `self._discover_features()` in `__init__` and additional checks later don't touch network again.  So asking for `self.supports_method` if an in-mem query. 

> (...) so a long-term strategy is needed.

I agree, that the way this lib is written is not ideal :) it grew organically from a weekend hack. While an implementation based fully on `desc_xml` would definitely be nicer underneath, most people seem to prefer look the other way and add bits and pieces here and there w/o attempting larger refactor :D


In other words: 
I can merge this as is if you wish. There are other places in this lib which also cause unexpected exceptions on some hardware, so no real harm is done by skipping checks here as well.

**Comment by isolationism** [comment]
> This is how self.supports_method(...) works underneath

Oh, that's excellent. My apologies, I so obviously didn't look at the code. In the middle of my workday struggling with Webpack right now so didn't look, but this actually sounds like the right move for the short term.

> most people seem to prefer to look the other way and add bits and pieces here and there w/o attempting larger refactor

Guilty as charged, but as I looked through the code I thought I could picture some ways it could be improved. I can't promise a timeline and I'm pretty busy with work these days, but this is one of the better Yamaha RX client libraries out there and it looks like Yahama is going to be sticking with the protocol for the forseeable future. It's probably worth the investment.

If you aren't in a rush let's leave this here for now and I'll check out the feature check API call and amend my commit when I get a chance.

**Comment by wuub** [comment]
No rush :) It's a slow moving project :)

Good luck with webpack

**Comment by wuub** [comment]
told ya it's a slow moving project ;)
